app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -36,6 +36,15 @@ def build_enumerator(_run, cursor:)
       collection_enum = case collection
       when ActiveRecord::Relation
         enumerator_builder.active_record_on_records(collection, cursor: cursor)
+      when ActiveRecord::Batches::BatchEnumerator
+        relation = collection.instance_variable_get(:@relation)
+        batch_size = collection.instance_variable_get(:@of)
```

Comment:
I think we want a test to ensure that a simple batch enumerator has those with the expected values. Additionally I think we should prevent the usage of `@start` and `@finish`, or support it properly

Reply:
Ah yes good call! Let's prevent usage, I think it would be tricky to support them since we're using the base relation here to build the enumerator, which doesn't factor in the `@start` / `@finish` values.